service/history/workflow/update/update.go

```diff
@@ -351,6 +356,31 @@ func (u *Update) onResponseMsg(
 	return nil
 }
 
+func (u *Update) cancel(
```

Comment:
It's not the end of the world but I had hoped that the only mutating interaction the state machine would have with the outside world would be via messages. What do you think about adding a generic protocol.Cancel message (or maybe protocol.Terminate) that all state machines would have to handle?